pkg/chunk/iterator.go

```diff
@@ -29,14 +29,18 @@ type LazySeriesIterator struct {
 }
 
 // NewLazySeriesIterator creates a LazySeriesIterator.
-func NewLazySeriesIterator(chunkStore *Store, metric model.Metric, from model.Time, through model.Time, matchers []*metric.LabelMatcher) *LazySeriesIterator {
+func NewLazySeriesIterator(chunkStore *Store, metric model.Metric, from model.Time, through model.Time, matchers []*metric.LabelMatcher) (*LazySeriesIterator, error) {
+	_, ok := metric[model.MetricNameLabel]
```

Comment:
Cache the metric name in the struct and remove the error case in `createSampleSeriesIterator`?